sdk/core/Azure.Core/src/ResourceIdentifier.cs

```diff
@@ -13,24 +13,49 @@ namespace Azure.Core
     /// </summary>
     public sealed class ResourceIdentifier : IEquatable<ResourceIdentifier>, IComparable<ResourceIdentifier>
     {
+        private enum SpecialType
+        {
+            Subscription,
+            ResourceGroup,
+            Location,
+            Provider
+        }
```

Comment:
If this enum has a 'None' entry then you don't need to box a null into it (or have a nullable property).

Reply:
I thought about that option, I guess I could perf test both and see if there is a difference

Reply:
this seems to be a noop on perf, so from a style perspective do we want SpecialType.None or null?

Reply:
I would prefer that we don't use the `Nullable<T>` for this enum and instead use None. @annelo-msft @KrzysztofCwalina thoughts?